src/v/cluster_link/topic_reconciler.h

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/dev/licenses/rcl.md
+ *
+ */
+
+#pragma once
+
+#include "cluster_link/deps.h"
+#include "kafka/data/rpc/deps.h"
+
+#include <seastar/core/gate.hh>
+
+namespace cluster_link {
+
+/**
+ * @brief Responsible for picking up updates to mirror topics and applying them
+ */
+class topic_reconciler {
```

Comment:
why not making it a task ?

I feel the generic mechanism of retriggering a task when link changes would be nice